ecl/hql/hqlgram.y

```diff
@@ -6495,6 +6495,14 @@ primexpr1
                         {
                             $$.setExpr(createValue(no_matched, makeBoolType(), $3.getExpr())); //, parser->createUniqueId()));
                         }
+    | MATCHED '(' dataRow ')'
+                        {
+                            $$.setExpr(createValue(no_matched_injoin, makeBoolType(), $3.getExpr()));
+                        }
+    | MATCHED '(' dataSet ')'
```

Comment:
MATCHED(ROWS(RIGHT))
@ghalliday removing this does cause a syntax error trying to MATCH(right) on a group denormalize dataset.  Not sure if there is something else I'm missing.